ironfish/src/wallet/accounts.test.slow.ts

```diff
@@ -125,6 +125,88 @@ describe('Accounts', () => {
+    // Initialize the database and chain
+    const strategy = nodeTest.strategy
+    const node = nodeTest.node
+    const chain = nodeTest.chain
+
+    const accountA = await node.accounts.createAccount('A', true)
+
+    // Initial balance should be 0
+    await node.accounts.updateHead()
+    await expect(node.accounts.getBalance(accountA)).resolves.toEqual({
+      confirmed: BigInt(0),
+      unconfirmed: BigInt(0),
+    })
+
+    // Create a block with a miner's fee awarded to account A
+    const minersfee = await strategy.createMinersFee(BigInt(0), 2, accountA.spendingKey)
+    const newBlock = await chain.newBlock([], minersfee)
+    const addResult = await chain.addBlock(newBlock)
+    expect(addResult.isAdded).toBeTruthy()
+
+    // Account A should now have a balance of 2000000000 after adding the miner's fee
+    await node.accounts.updateHead()
+    await expect(node.accounts.getBalance(accountA)).resolves.toEqual({
+      confirmed: BigInt(2000000000),
+      unconfirmed: BigInt(2000000000),
+    })
+
+    // Create a second account
+    const accountB = await node.accounts.createAccount('B', true)
+
+    // Load account data from db
+    await node.accounts.loadAccountsFromDb()
+
+    // Account A should have one transaction
+    expect(Array.from(accountA.getTransactions()).length).toEqual(1)
+
+    // Account B should have zero transactions
+    expect(Array.from(accountB.getTransactions()).length).toEqual(0)
+  }, 600000)
+
+  it('Loads only the nullifiers that an account owns a note for', async () => {
+    // Initialize the database and chain
+    const strategy = nodeTest.strategy
+    const node = nodeTest.node
+    const chain = nodeTest.chain
+
+    const accountA = await node.accounts.createAccount('A', true)
+
+    // Initial balance should be 0
+    await node.accounts.updateHead()
+    await expect(node.accounts.getBalance(accountA)).resolves.toEqual({
+      confirmed: BigInt(0),
+      unconfirmed: BigInt(0),
+    })
+
+    // Create a block with a miner's fee awarded to account A
+    const minersfee = await strategy.createMinersFee(BigInt(0), 2, accountA.spendingKey)
+    const newBlock = await chain.newBlock([], minersfee)
+    const addResult = await chain.addBlock(newBlock)
+    expect(addResult.isAdded).toBeTruthy()
+
+    // Account A should now have a balance of 2000000000 after adding the miner's fee
+    await node.accounts.updateHead()
+    await expect(node.accounts.getBalance(accountA)).resolves.toEqual({
+      confirmed: BigInt(2000000000),
+      unconfirmed: BigInt(2000000000),
+    })
+
+    // Create a second account
+    const accountB = await node.accounts.createAccount('B', true)
+
+    // Load account data from db
+    await node.accounts.loadAccountsFromDb()
+
+    // Account A should have one nullifier
+    expect(accountA['nullifierToNoteHash'].size).toEqual(1)
+
+    // Account B should have zero nullifiers
+    expect(accountB['nullifierToNoteHash'].size).toEqual(0)
```

Comment:
If we put these lines before `loadAccountsFromDb` it'll still pass. So you can delete `loadAccountsFromDb` and the test will still pass. How do we know we've prevented the bug?

Reply:
This test fails on `accounts-performance` where `loadAccountsFromDb` is loading nullifiers into both accounts.

I'll update the test to clear the in-memory stores for both accounts before loading so that it's clearer that the load is changing for both accounts